crates/btc-relay/src/types.rs

```diff
@@ -45,3 +49,85 @@ impl<AccountId, BlockNumber> RichBlockHeader<AccountId, BlockNumber> {
         })
     }
 }
+
+pub struct OpReturnPaymentData<T: frame_system::Config> {
```

Comment:
Not sure that containing this logic adds much over the previous approach

Reply:
The advantage is that the theft check and transaction validation logic is now shared, meaning there are no discrepancies between the two.